src/stan/mcmc/hmc/base_hmc.hpp

```diff
@@ -71,6 +71,7 @@ class base_hmc : public base_mcmc {
     struct_writer.begin_record();
     struct_writer.write("stepsize", get_nominal_stepsize());
     struct_writer.write("inv_metric", z_.inv_e_metric_);
+    struct_writer.write("metric_type", z_.metric_type());
```

Comment:
This is a nitpick, but could this be written out before the actual metric itself? I'm imaging a situation where you're reading this file and want to pre-allocate something for the metric to be read into, but that requires knowing the type before you start reading it